File: advent1.py
list1 = []
list2 = []

with open("advent1.in") as file:
    for line in file:
        parts = line.split(' ')
        list1.append(int(parts[0]))
        list2.append(int(parts[-1]))

similarities = 0
for l1 in list1:
    similarities += l1 * list2.count(l1)

print(similarities)

File: advent1.jl
using DataStructures

number_regex = r"^(\d+)\s+(\d+)$"
result = 0
left = []
right = []

open("data/advent1.in") do file
    while !eof(file)
        line = readline(file)
        matches = match(number_regex, line)

        push!(left, parse(Int, matches[1]))
        push!(right, parse(Int, matches[2]))
    end
end

right_counter = counter(right)

for l in left
    global result += l * right_counter[l]
end

println(result)
